### PR TITLE
[AArch64] Add assembly/disassembly for {S,U,SU,US}TMOPA instructions

### PR DESCRIPTION
**Pull request by momchil-velikov**
The new instructions are described in
https://developer.arm.com/documentation/ddi0602/2024-09/SME-Instructions

Co-Authored-By:  Marian Lukac <Marian.Lukac@arm.com>

### CODE REVIEW

**Review by jthackray — APPROVED** [review]
LGTM

File: llvm/test/MC/AArch64/SME2p2/utmopa-diagnostics.s [hunk 1]
```diff
@@ -0,0 +1,77 @@
+// RUN: not llvm-mc -triple=aarch64 -show-encoding -mattr=+sme2p2 2>&1 < %s| FileCheck %s
+
+// --------------------------------------------------------------------------//
+// Invalid ZA register
+
+sutmopa  za4.s, {z30.b-z31.b}, z31.b, z31[3]
```
Comment (SpencerAbson) [inline llvm/test/MC/AArch64/SME2p2/utmopa-diagnostics.s:6]: I think this should have contained the tests for `utmopa`, we have these in `sutmopa-diagnostics.s`.

File: llvm/test/MC/AArch64/SME2p2/stmopa-diagnostics.s [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+stmopa  za3.s, {z29.h-z30.h}, z31.h, z31[3]
+// CHECK: [[@LINE-1]]:{{[0-9]+}}: error: Invalid vector list, expected list with 2 consecutive SVE vectors, where the first vector is a multiple of 2 and with matching element types
+// CHECK-NEXT: stmopa  za3.s, {z29.h-z30.h}, z31.h, z31[3]
+// CHECK-NOT: [[@LINE-1]]:{{[0-9]+}}:
+
+// --------------------------------------------------------------------------//
+// Invalid ZK register
+
+stmopa  za3.s, {z28.b-z29.b}, z31.b, z27[3]
```
Comment (SpencerAbson) [inline llvm/test/MC/AArch64/SME2p2/stmopa-diagnostics.s:32]: We could include:
`stmopa  za3.s, {z28.b-z29.b}, z31.b, z19[3]`/`za3.s, {z28.h-z29.h}, z31.h, z19[3]`.

File: llvm/test/MC/AArch64/SME2p2/sutmopa-diagnostics.s [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+sutmopa  za3.s, {z28.b-z29.b}, z31.b, z27[2]
+// CHECK: [[@LINE-1]]:{{[0-9]+}}: error: invalid restricted vector register, expected register in z20..z23 or z28..z31
+// CHECK-NEXT: sutmopa  za3.s, {z28.b-z29.b}, z31.b, z27[2]
+// CHECK-NOT: [[@LINE-1]]:{{[0-9]+}}:
+
+// --------------------------------------------------------------------------//
+// Invalid immediate
+
+
```
Comment (SpencerAbson) [inline llvm/test/MC/AArch64/SME2p2/sutmopa-diagnostics.s:40]: nit: blank-lines

File: llvm/test/MC/AArch64/SME2p2/ustmopa-diagnostics.s [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+ustmopa  za3.s, {z28.b-z29.b}, z31.b, z27[2]
+// CHECK: [[@LINE-1]]:{{[0-9]+}}: error: invalid restricted vector register, expected register in z20..z23 or z28..z31
+// CHECK-NEXT: ustmopa  za3.s, {z28.b-z29.b}, z31.b, z27[2]
+// CHECK-NOT: [[@LINE-1]]:{{[0-9]+}}:
+
+// --------------------------------------------------------------------------//
+// Invalid immediate
+
+
```
Comment (SpencerAbson) [inline llvm/test/MC/AArch64/SME2p2/ustmopa-diagnostics.s:40]: nit: blank-lines

File: llvm/test/MC/AArch64/SME2p2/stmopa-diagnostics.s (hunk 2 above)
Comment (momchil-velikov) [reply llvm/test/MC/AArch64/SME2p2/stmopa-diagnostics.s:41]: Done.

File: llvm/test/MC/AArch64/SME2p2/ustmopa-diagnostics.s (hunk 4 above)
Comment (momchil-velikov) [reply llvm/test/MC/AArch64/SME2p2/ustmopa-diagnostics.s]: Done.

File: llvm/test/MC/AArch64/SME2p2/sutmopa-diagnostics.s (hunk 3 above)
Comment (momchil-velikov) [reply llvm/test/MC/AArch64/SME2p2/sutmopa-diagnostics.s]: Done

**Review by SpencerAbson — APPROVED** [review]
LGTM, thanks.

**Pull request merged**